tools/ak3-core.sh

```diff
@@ -437,9 +437,14 @@ flash_generic() {
     if [ ! "$imgblock" ]; then
       abort "$1 partition could not be found. Aborting...";
     fi;
-    # TODO: add dynamic partition resizing using lptools_static instead of aborting
     if [ "$(wc -c < $img)" -gt "$(wc -c < $imgblock)" ]; then
-      abort "New $1 image larger than $1 partition. Aborting...";
+      [ $path == /dev/block/mapper ] || abort "New $1 image larger than $1 partition. Aborting...";
+      islp=true;
+      $bin/lptools remove $1_ak3 || abort "Removing $1_ak3 failed. Aborting...";
```

Comment:
Won't this abort always trigger since (unless something has failed previously) $1_ak3 should never exist?

Reply:
The exit status is 0, even if the partition doesn't exist. The only thing that causes it to return a different exit status is if it fails to write the partition table after the operation (which, unless I'm missing something, would be totally unrelated to the remove operation). This is consistent with how it is handled in `fastboot`/`fs_mgr`.

Reply:
Hmm but if there's nothing to remove then shouldn't there be nothing to change in the partition table?

Reply:
The `replace` operation also removes the source partition, so it should never exist at that point. I assume it was there in `lp-self-flash` to clean up in the case of a failure after the initial `create` that prevented `replace` from running.

Reply:
Right, so the remove line should probably stay in case that could happen for AK3 as well, but the || abort should probably be removed on that line. Any potential table writing issue will still get caught by the create || abort line.